### ISSUE TITLE
0.49.0 breaks code when conditionally returning boolean values in functions

### ISSUE THREAD

**Issue by swernerx** [rollup/rollup#1588]
Rollup is breaking my [prepublish](https://github.com/sebastian-software/prepublish/) tool with the newest 0.49 - this is probably related to the new tree shaking which might be a little too intensive ;)

Input:

```js
export function helper(value) {
  if (value) {
    return true
  }

  somethingElse()
}
```

Output:

```js
function helper(value) {
  somethingElse();
}

export { helper };
```

[REPL Link](https://rollupjs.org/repl?version=0.49.0&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGZ1bmN0aW9uJTIwaGVscGVyKHZhbHVlKSUyMCU3QiU1Q24lMjAlMjBpZiUyMCh2YWx1ZSklMjAlN0IlNUNuJTIwJTIwJTIwJTIwcmV0dXJuJTIwdHJ1ZSU1Q24lMjAlMjAlN0QlNUNuJTVDbiUyMCUyMHNvbWV0aGluZ0Vsc2UoKSU1Q24lN0QlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

It seems to affect boolean returns somehow... problem goes away when executing some other function.

**Comment by Rich-Harris** [comment]
[Fixed in 0.49.1](https://rollupjs.org/repl?version=0.49.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGZ1bmN0aW9uJTIwaGVscGVyKHZhbHVlKSUyMCU3QiU1Q24lMjAlMjBpZiUyMCh2YWx1ZSklMjAlN0IlNUNuJTIwJTIwJTIwJTIwcmV0dXJuJTIwdHJ1ZSU1Q24lMjAlMjAlN0QlNUNuJTVDbiUyMCUyMHNvbWV0aGluZ0Vsc2UoKSU1Q24lN0QlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyY2pzJTIyJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=) 😀

**Comment by dherges** [comment]
@Rich-Harris @lukastaegert and all the others involved: Thanks for the quick work to get the regressions fixed!

I think I hit the #1585 and #1588 and can confirm that it is now working again!
